xapian-letor/tests/api_letor.cc

```diff
@@ -1006,14 +1010,14 @@ DEFINE_TESTCASE(createfeaturevector_idffeature, generated)
     Xapian::MSet mset;
     mset = enquire.get_mset(0, 10);
 
-    TEST(!mset.empty());
-
     Xapian::QueryParser queryparser;
     queryparser.set_stemmer(Xapian::Stem("en"));
     queryparser.set_stemming_strategy(queryparser.STEM_ALL_Z);
     queryparser.add_prefix("title", "S");
     queryparser.add_prefix("description", "XD");
 
+    // As the feature values depend on "title","body" and "whole",
+    // we need to seperate it out instead of just writing Query("score").
```

Comment:
I still had to check the `TfFeature` implementation. Can we maybe say "as the feature values depend on different prefixed terms in the query" or something like that?

Also, it's "separate" not "seperate" (which is causing the CI failure, since we have a check for that in `xapian-check-patch` — it's a good habit to run this yourself every so often because a CI run takes a fair amount of time, so fixing these problems before you push makes everything smoother).

Reply:
I did run it before pushing it but I failed to realize I was running it before pushing the changes and hence it was showing the results of the previous run and so I was not able to catch it.
Fixed in a fixup commit.

Reply:
It just takes a diff as input, so you can get it to check uncommitted changes like so:

```
git diff|xapian-maintainer-tools/xapian-check-patch
```

Or `git diff --cached|...` if you want to check changes staged for commit, or `git show SHA|...` if you want to check the changes in the single commit with sha `SHA`, and so on.